corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js

```diff
@@ -69,23 +70,43 @@ hqDefine("cloudcare/js/form_entry/utils", function () {
     /**
      * Sets a div to be a mapbox geocoder input
      * @param {(string|string[])} divId - Div ID for the Mapbox input
-     * @param {Object} itemCallback - function to call back after new search
-     * @param {Object} clearCallBack - function to call back after clearing the input
+     * @param {function} itemCallback - function to call back after new search
+     * @param {function} clearCallBack - function to call back after clearing the input
      * @param {Object} initialPageData - initial_page_data object
      * @param {function|undefined} inputOnKeyDown - inputOnKeyDown function (optional)
+     * @param {boolean} showGeolocationButton - show geolocation button. Defaults to false. (optional)
+     * @param {boolean} geolocateOnLoad - geolocate the user's location on load. Defaults to false. (optional)
+     * @param {boolean} setProximity - set proximity to user's location. Defaults to false. (optional)
      */
-    module.renderMapboxInput = function (divId, itemCallback, clearCallBack, initialPageData, inputOnKeyDown) {
+    module.renderMapboxInput = function (
+        divId,
+        itemCallback,
+        clearCallBack,
+        initialPageData,
+        inputOnKeyDown,
+        showGeolocationButton= false,
+        geolocateOnLoad = false,
+        setProximity = false
+    ) {
+        showGeolocationButton = showGeolocationButton || toggles.toggleEnabled('GEOCODER_MY_LOCATION_BUTTON');
+        geolocateOnLoad = geolocateOnLoad || toggles.toggleEnabled('GEOCODER_AUTOLOAD_USER_LOCATION');
+        setProximity = setProximity || toggles.toggleEnabled('GEOCODER_USER_PROXIMITY');
```

Comment:
What's the idea with parameterizing these flags?  That certain instantiations might have them always-on?  Or is that just to make it easier to test in the future?

Reply:
Yea, it mostly for future.